File: InsertionSort.js
//InsertionSort
function insertionSort(array){
    for(let i = 1; i < array.length; i++){
        //Armazenar elemento do array desordenado
        let key = array[i];
        let j = i - 1
        while((j > -1) && (key < array[j])){
            array[j+1] = array[j];
            j--;
        }
        array[j+1] = key;
    }

    return array;
}

let array = [6,5,1,8,7,2,3,4];
insertionSort(array);
console.log(array);

File: InsertionSort.py
#InsertionSort
def insertion_sort(array):
    for step in range(1, len(array)):
        key = array[step]
        j = step - 1
        
        #Comparar key com cada elemento na esquerda até que um elemento menor que key seja encontrado
        while j >= 0 and key < array[j]:
            array[j + 1] = array[j]
            j = j - 1

        #Posicionar o menor valor:
        array[j + 1] = key

data = [6,3,5,1,7,2,8,4]
insertion_sort(data)
print('\nArray em ordem crescente:')
print(data)


